internal/tiltfile/tiltfile_test.go

```diff
@@ -4798,6 +4798,48 @@ k8s_resource('foo', objects=['baz:namespace:qux'])
 	f.loadErrString("No object identified by the fragment \"baz:namespace:qux\" could be found. Possible objects are: \"foo:Deployment:default\", \"baz:Namespace:default\"")
 }
 
+func TestK8sResouceObjectsNonWorkloadOnly(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("namespace.yaml", namespace("baz"))
+
+	f.file("Tiltfile", `
+k8s_yaml('secret.yaml')
+k8s_yaml('namespace.yaml')
+k8s_resource(new_name='foo', objects=['bar', 'baz:namespace:default'])
+`)
+
+	f.load()
+
+	f.assertNextManifest("foo", k8sObject("bar", "Secret"), k8sObject("baz", "Namespace"))
+	f.assertNoMoreManifests()
+}
+
+func TestK8sWorkloadOnlyResourceWithAllTheOptions(t *testing.T) {
```

Comment:
do you have a comparable test for a non-workload resource with all the options? would be good to make sure that that codepath stays intact